lib/puma/server.rb

```diff
@@ -363,7 +363,7 @@ def handle_servers
                     end
 
                     pool << client
-                    pool.wait_until_not_full unless queue_requests
+                    pool.wait_until_not_full
```

Comment:
Was there a reason `queue_requests` was removed here?

Reply:
I think the reason is explained adequately in the OP?